### ISSUE TITLE
Add ability to state only which type a function call argument should be

### ISSUE THREAD

**Issue by Jameskmonger** [alsatian-test/alsatian#154]
I want to check that a function was called with some arguments, but I don't care about all of them.

If I have this interface:

```
interface Chef {
    cook(name: string, calories?: number): void;
}
```

And I want to test that `cook` is called with the name "Meatball Magic" and **any** calorie amount, I can't do that.

This fails the assertion:

```
Expect(someChefInstance.cook).toHaveBeenCalledWith("Meatball Magic");
```

As does this:

```
Expect(someChefInstance.cook).toHaveBeenCalledWith("Meatball Magic", undefined);
```

**Comment by jamesrichford** [comment]
Add an Any function that takes a type or can be passed in itself to indicate an argument at this index can be anything

**Comment by jamesrichford** [comment]
@Jameskmonger - hey mate, if you're looking into this I'd hold off as I'm reworking the toHaveBeenCalledWith / toHaveBeenCalled functions which will probably cause a bit of headache here. Sorry mate

**Comment by Jameskmonger** [comment]
No worries @jamesrichford I hadn't started yet :P

**Comment by jamesrichford** [comment]
closed by #190
